api/app/src/providers/oauth2.ts

```diff
@@ -16,6 +16,7 @@ export const oauthUserTokenResponse = z.object({
 
 export interface OAuth2 {
   getAuthUri(state: string): Promise<string>;
+  revokeProviderAccess(connectedUser: ConnectedUser): void;
```

Comment:
I thought the compiler would complain, but even if it doesn't I think its worth to make the return type a Promise, since the default implementation has it, and so any other implementation would follow suit?